src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ticketRuleModal.tsx

```diff
@@ -73,158 +90,100 @@ class TicketRuleModal extends React.Component<Props, State> {
-            set(newState, `dynamicFieldChoices[${field.name}]`, dynamicFieldChoices);
-            return newState;
-          });
-          resolve(result);
-        }
-      });
+  updateDynamicFieldChoices = (field: IssueConfigField, result: any): void => {
+    const dynamicFieldChoices = result.options.map(obj => [obj.value, obj.label]);
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      set(newState, `dynamicFieldChoices[${field.name}]`, dynamicFieldChoices);
+      return newState;
     });
+  };
 
-  getFieldProps = (field: IssueConfigField) =>
-    field.url
-      ? {
-          loadOptions: (input: string) => this.getOptions(field, input),
-          async: true,
-          cache: false,
-          onSelectResetsInput: false,
-          onCloseResetsInput: false,
-          onBlurResetsInput: false,
-          autoload: true,
-        }
-      : {};
-
-  addFields = (fields: FormField[]): void => {
-    const title = {
-      name: 'title',
-      label: 'Title',
-      type: 'string',
-      default: 'This will be the same as the Sentry Issue.',
-      disabled: true,
-    };
-    const description = {
-      name: 'description',
-      label: 'Description',
-      type: 'string',
-      default: 'This will be generated from the Sentry Issue details.',
-      disabled: true,
+  getFormProps = (): Form['props'] => {
+    const {closeModal} = this.props;
+
+    return {
+      ...this.getDefaultFormProps(),
+      cancelLabel: t('Close'),
+      onCancel: closeModal,
+      onSubmit: this.onFormSubmit,
+      submitLabel: t('Apply Changes'),
     };
-    fields.unshift(description);
-    fields.unshift(title);
   };
 
-  render() {
-    const {Header, Body, formFields, closeModal, link, ticketType, instance} = this.props;
-
-    const text = t(
-      'When this alert is triggered %s will be created with the following fields. ',
-      ticketType
+  /**
+   * Set the initial data from the Rule, replace `title` and `description` with
+   * disabled inputs, and use the cached dynamic choices.
+   */
+  cleanFields = (): IssueConfigField[] => {
+    const {formFields, instance} = this.props;
+
+    return [
+      {
+        name: 'title',
+        label: 'Title',
+        type: 'string',
+        default: 'This will be the same as the Sentry Issue.',
+        disabled: true,
+      } as IssueConfigField,
+      {
+        name: 'description',
+        label: 'Description',
+        type: 'string',
+        default: 'This will be generated from the Sentry Issue details.',
+        disabled: true,
+      } as IssueConfigField,
+    ].concat(
+      Object.values(formFields)
+        .filter(f => !['title', 'description'].includes(f.name))
+        .map(field => {
+          if (instance.hasOwnProperty(field.name)) {
+            field.default = instance[field.name];
+          }
+          if (
+            ['assignee', 'reporter'].includes(field.name) &&
```

Comment:
@mgaeta hmmm trying to understand what's special about these. Is it because they could have saved default values? That can happen for other fields as well.